src/HomogeneousCalorimeter_geo.cpp

```diff
@@ -288,311 +177,181 @@ std::tuple<Volume, Position> build_module(Detector& desc, xml::Collection_t& plm
-
-  // line placement
-  int mid = 0;
-  for (xml::Collection_t pl(plm, _Unicode(line)); pl; ++pl) {
-    Position pos(dd4hep::getAttrOrDefault<double>(pl, _Unicode(x), 0.),
-                 dd4hep::getAttrOrDefault<double>(pl, _Unicode(y), 0.),
-                 dd4hep::getAttrOrDefault<double>(pl, _Unicode(z), 0.));
-    Position rot(dd4hep::getAttrOrDefault<double>(pl, _Unicode(rotx), 0.),
-                 dd4hep::getAttrOrDefault<double>(pl, _Unicode(roty), 0.),
-                 dd4hep::getAttrOrDefault<double>(pl, _Unicode(rotz), 0.));
-    // determine axis
-    std::string axis = dd4hep::getAttrOrDefault<std::string>(pl, _Unicode(axis), "x");
-    std::transform(axis.begin(), axis.end(), axis.begin(), [](char c) { return std::tolower(c); });
-    if ((axis != "x") && (axis != "y") && (axis != "z")) {
-      std::cerr << "HomogeneousCalorimeter Error: cannot determine axis of line " << axis
-                << ", abort placement of this line." << std::endl;
-      continue;
-    }
+  pos = 0;
+  while ((pos = iposy.find(delimiter)) != std::string::npos) {
+    token = iposy.substr(0, pos);
+    inner_outer_posy.push_back(atof(token.c_str()));
+    iposy.erase(0, pos + delimiter.length());
+  }
+  inner_outer_posy.push_back(atof(iposy.c_str()));
 
-    int begin = dd4hep::getAttrOrDefault<int>(pl, _Unicode(begin), 0);
-    int nmods = pl.attr<int>(_Unicode(nmods));
+  for (auto& p : points) {
+    if (p.x() < min_ptsx)
+      min_ptsx = p.x();
+    if (p.x() > max_ptsx)
+      max_ptsx = p.x();
+    if (p.y() < min_ptsy)
+      min_ptsy = p.y();
+    if (p.y() > max_ptsy)
+      max_ptsy = p.y();
+  }
 
-    std::vector<Position> trans;
-    for (int i = 0; i < nmods; ++i) {
-      Position tran{(axis == "x") ? pos.x() + (begin + i) * modSize.x() : pos.x(),
-                    (axis == "y") ? pos.y() + (begin + i) * modSize.y() : pos.y(),
-                    (axis == "z") ? pos.z() + (begin + i) * modSize.z() : pos.z()};
-      trans.push_back(tran);
-    }
 
-    // mirror placement
-    if (mirrorx) {
-      size_t curr_size = trans.size();
-      for (size_t i = 0; i < curr_size; ++i) {
-        trans.push_back(Position{-trans[i].x(), trans[i].y(), trans[i].z()});
-      }
-    }
-    if (mirrory) {
-      size_t curr_size = trans.size();
-      for (size_t i = 0; i < curr_size; ++i) {
-        trans.push_back(Position{trans[i].x(), -trans[i].y(), trans[i].z()});
-      }
-    }
-    if (mirrorz) {
-      size_t curr_size = trans.size();
-      for (size_t i = 0; i < curr_size; ++i) {
-        trans.push_back(Position{trans[i].x(), trans[i].y(), -trans[i].z()});
-      }
-    }
+  // Place the modules, use row and column ID for each modules
+  // row and column ID start from the top left corner
+  //
+  int row = 0, column = 0;
+  for (auto& p : points) {
+    column = (p.x() - min_ptsx) / modSize.x();
+    row = (max_ptsy - p.y()) / modSize.y();
```

Comment:
Would be nice to configure cells starting from integer row/column pairs and then multiplying by modSize. If you insist on doing it backwards, at least let's
```suggestion
    column = std::round((p.x() - min_ptsx) / modSize.x());
    row = std::round((max_ptsy - p.y()) / modSize.y());
```

Reply:
I started from the position as there is a function to fill the circle with box, which will output the positions of boxes.

Reply:
solved in the last commit